bottlecap/src/lifecycle/invocation/triggers/sqs_event.rs

```diff
@@ -198,6 +202,81 @@ impl Trigger for SqsRecord {
     }
 }
 
+// extractTraceContextfromAWSTraceHeader extracts trace context from the
+// AWSTraceHeader directly. Unlike the other carriers in this file, it should
+// not be passed to the tracer.Propagator, instead extracting context directly.
+pub(crate) fn extract_trace_context_from_aws_trace_header(
+    value: Option<String>,
+) -> Result<SpanContext, String> {
+    if value.is_none() {
+        return Err("awstrace_header is empty".to_string());
+    }
+    let value = value.unwrap();
+    if !value.starts_with("Root=") {
+        return Err("awstrace_header mismatch".to_string());
+    }
+
+    let mut start_part = 0;
+    let mut trace_id = String::new();
+    let mut parent_id = String::new();
+    let mut sampled = String::new();
+
+    let length = value.len();
+    while start_part < length {
+        let end_part = value[start_part..]
+            .find(';')
+            .map_or(length, |i| i + start_part);
+        let part = &value[start_part..end_part];
+
+        if part.starts_with("Root=") {
+            if trace_id.is_empty() {
+                trace_id = part[24..].to_string();
+            }
+        } else if part.starts_with("Parent=") {
+            if parent_id.is_empty() {
+                parent_id = part[7..].to_string();
+            }
+        } else if part.starts_with("Sampled=") && sampled.is_empty() {
+            sampled = part[8..].to_string();
+        }
+
+        if !trace_id.is_empty() && !parent_id.is_empty() && !sampled.is_empty() {
+            break;
+        }
+        start_part = end_part + 1;
+    }
```

Comment:
Can't we just do a simpler algo?
```Root=1-64cc2edd-112fbf1701d1355973a11d57;Parent=7d5a9776024b2d42;Sampled=0```
Split by `;`
- `Root=1-64cc2edd-112fbf1701d1355973a11d57`
- `Parent=7d5a9776024b2d42`
- `Sampled=0`

Then split by `=`?

Reply:
I chatgpt-translated the algorithm used in go.
I am not sure if there are side effects of using an easier split. I would argue a regexp would be much better and faster, but can you confirm that the format is a `;` concatenations of those 3 headers?

Reply:
the logic seems to be to parse all the `;` separated segments until all 3 variables are set. I did not want to spend too much time thinking on side effects